cosmwasm/contracts/crosschain-swaps/src/checks.rs

```diff
@@ -31,46 +31,45 @@ fn validate_explicit_receiver(receiver: &str) -> Result<(String, Addr), Contract
             receiver: receiver.to_string(),
         });
     }
-    let channel_id = &channel[8..];
-    if channel_id.is_empty() || channel_id.parse::<u64>().is_err() {
-        return Err(ContractError::InvalidReceiver {
-            receiver: receiver.to_string(),
-        });
-    };
 
     let Ok(_) = bech32::decode(&address) else {
         return Err(ContractError::InvalidReceiver { receiver: receiver.to_string() })
     };
 
-    Ok((channel.to_string(), Addr::unchecked(address)))
+    let registry = Registry::default(deps);
```

Comment:
Did we decide how we want to set the address for the default value yet?

Reply:
decided, no. But I added it as a compilation flag in this PR. So you can compile with the env variable `REGISTRY_CONTRACT` and that will set it. With the optimizer:

```
docker run --rm -v "$(pwd)":/code -e REGISTRY_CONTRACT=osmo14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9sq2r9g9 \
  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
  cosmwasm/workspace-optimizer-arm64:0.12.11
  ```